File: N3.15/N3.15.swift
import Foundation

func reglamentCheck(num: Int) -> Bool {
    var prevDigit = num % 10
    var num = num / 10
    while num != 0 {
        let currentDigit = num % 10
        if currentDigit <= prevDigit {
            prevDigit = currentDigit
            num /= 10
        } else {
            return false
        }
    }
    return true
}

func numReverser(numblank: Int) -> Int {
    if reglamentCheck(num: numblank) {
        var reversedNum = 0
        var numblank = numblank
        while numblank != 0 {
            reversedNum = reversedNum * 10 + numblank % 10
            numblank /= 10
        }
        return reversedNum
    } else {
        return 0
    }
}

print("Enter the number of numbers : ")
if let n = Int(readLine()!), n > 0 {
    print("Enter \(n) numbers:")
    for _ in 0..<n {
        if let number = Int(readLine()!) {
            print(numReverser(numblank: number), terminator: " ")
        }
    }
    print()
}

File: N3.15/N3.15.py
def reglament_check(num):
    prev_digit = num % 10
    num = num // 10
    while num != 0:
        current_digit = num % 10
        if current_digit <= prev_digit:
            prev_digit = current_digit
            num //= 10
        else:
            return False
    return True

def num_reverser(num):
    if reglament_check(num):
        reversed_num = 0
        while num != 0:
            reversed_num = reversed_num * 10 + num % 10
            num //= 10
        return reversed_num
    else:
        return 0

# Заранее заданные числа
numbers = [1234, 4321, 9876, 6789]

for number in numbers:
    print(num_reverser(number), end=" ")
print()